File: backend-v2/api/schemas/quib.py
from ninja import ModelSchema

from quib.models import Quib
from quiblet.models import Quiblet
from user.models import Profile


class QuibQuibletSchema(ModelSchema):
    class Meta:
        model = Quiblet
        fields = ["id", "name", "avatar"]


class QuibPosterSchema(ModelSchema):
    class Meta:
        model = Profile
        fields = ["id", "username", "avatar"]


class QuibSchema(ModelSchema):
    quiblet: QuibQuibletSchema
    poster: QuibPosterSchema

    class Meta:
        model = Quib
        fields = [
            "id",
            "slug",
            "title",
            "quiblet",
            "poster",
            "is_highlighted",
            "is_published",
            "cover",
            "cover_small",
            "content",
        ]

File: frontend-v2/types/quib.ts
import type { Nullable } from "./generics";

export interface Quib {
  upvotes: number;
  downvotes: number;
  user_vote_value: Nullable<number>;
  quiblet: {
    id: number;
    name: string;
    avatar: Nullable<string>;
  };
  poster: {
    id: number;
    username: string;
    name: Nullable<string>;
    avatar: Nullable<string>;
  };
  id: string;
  slug: string;
  title: string;
  cover: Nullable<string>;
  cover_small: Nullable<string>;
  content: Nullable<string>;
  created_at: string;
}
